### PR TITLE
a new VM 'proxy-server' in a vagrant stack

### PR DESCRIPTION
**Pull request by jstavel**
There is an authorization proxy server in Vagrantfile. It can be used for testing purposes mainly.

```shell
cd src/subscription-manager
vagrant up proxy-server
```
See README.md for details.

### CODE REVIEW

**Comment by jirihnidek** [comment]
@candlepin-pull-request-bot retest this please

File: Vagrantfile [hunk 1]
```diff
@@ -131,3 +131,27 @@ Vagrant.configure("2") do |config|
     }
   end
 end
+
+## VM for proxy server
+Vagrant.configure("2") do |config|
+  config.vm.hostname = 'proxy-server.subman.example.com'
+  config.vm.box = 'centos/7'
+  config.vm.define 'proxy-server', autostart: false
+
```
Comment (jirihnidek) [inline Vagrantfile]: Created VM has two NIC (one of them uses private network). It should have only one NIC using public network. Look at definition of VMs in #1786 and obsolete comments from Kevin in this PR.

File: vagrant/roles/proxy-server/files/squid.conf [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+acl localnet src 10.0.0.0/8		# RFC 1918 local private network (LAN)
+acl localnet src 100.64.0.0/10		# RFC 6598 shared address space (CGN)
+acl localhet src 169.254.0.0/16 	# RFC 3927 link-local (directly plugged) machines
+acl localnet src 172.16.0.0/12		# RFC 1918 local private network (LAN)
+acl localnet src 192.168.0.0/16		# RFC 1918 local private network (LAN)
+acl localnet src fc00::/7       	# RFC 4193 local private network range
+acl localnet src fe80::/10      	# RFC 4291 link-local (directly plugged) machines
+
+acl SSL_ports port 443
+acl SSL_ports port 8443
```
Comment (jirihnidek) [inline vagrant/roles/proxy-server/files/squid.conf]: Nitpick: add note about candlepin server using port 8443

File: vagrant/vagrant_proxy_server.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+- hosts: proxy-servers
+  roles:
+    - proxy-server
+
```
[the remark this answers is not in the document]
Comment (jstavel) [reply vagrant/vagrant_proxy_server.yml]: I do not understand this comments. There are variables in defaults/main.yml - proxyuser, password, ...

**Comment by jstavel** [comment]
I have merged the latest master - pxe-client was added. 
I have added a commentary for candlepin test port in squid.conf
I have solved a problem that hostmanager did not update /etc/hosts with IPs from ssh config. It used IP address of pxe-server for all IPs.

**Comment by jirihnidek** [comment]
@candlepin-pull-request-bot retest this please

**Comment by jstavel** [comment]
I've squashed the commits. Tried, Tested, Trusted.

**Comment by jstavel** [comment]
a section of vars in the file vagrant_proxy_server.yml:

```
  vars:
    htpasswd:
      name: squid
      path: /etc/squid/passwd
      owner: root
      group: squid
      mode: "0640"
      users:
        - { name: proxyuser, password: password }
```
is used for creating of htpasswd file for authorized access to the proxy server.
I may misunderstood something but I think this section is important for this ansible role.

**Comment by jstavel** [comment]
I have removed vars section in `vagrant/vagrant_proxy_server.yml`. The right variables are stored in `vagrant/roles/proxy-server/default/main.yml`.

**Pull request merged**